src/gui/tray/ColorSvgImageProvider.cpp

```diff
@@ -0,0 +1,68 @@
+#include <QPainter>
+#include <QLoggingCategory>
+#include <QEventLoop>
+#include <QNetworkAccessManager>
+#include <QPalette>
+#include <QSvgRenderer>
+#include <QNetworkReply>
+
+#include "ColorSvgImageProvider.h"
+
+namespace OCC {
+
+Q_LOGGING_CATEGORY(lcColorSvgImageProvider, "nextcloud.gui.tray.colorsvgimageprovider", QtInfoMsg)
+
+ColorSvgImageProvider::ColorSvgImageProvider()
+    : QQuickImageProvider(QQuickImageProvider::Image)
+{
+}
+
+
+QImage ColorSvgImageProvider::requestImage(const QString &id, QSize *size, const QSize &requestedSize)
+{
+    // TODO: Clean up
+    if (size) {
+        *size = requestedSize;
+    }
+    qCDebug(lcColorSvgImageProvider) << "Load" << id;
+    QSvgRenderer svgRenderer;
+    if (id.startsWith("data")) {
+        const QString prefix("data:image/svg+xml;utf8,");
+        Q_ASSERT(id.startsWith(prefix));
+        const auto svgData = QUrl::fromPercentEncoding(id.toLocal8Bit()).mid(prefix.size()).toLocal8Bit();
+        Q_ASSERT(svgRenderer.load(svgData));
+    } else if (id.startsWith("http")) {
+        const auto url = QUrl::fromUserInput(id);
+        auto networkAccessManager = new QNetworkAccessManager();
+        auto reply = networkAccessManager->get(QNetworkRequest(url));
+
+        // TODO: This loop is a little bit ugly but had no better idea yet to do synchronious network request
```

Comment:
Yeah somehow expected http wouldn't get in the way this time... looks like I've been too optimistic on that one. Which ones are going through HTTP?

But then... there is QQuickAsyncImageProvider you could switch to it, it's a bit more involved:
https://doc.qt.io/qt-5/qquickasyncimageprovider.html

Reply:
Thanks I look into that. The app icons get queried over http.